### DISCUSSION TITLE
Apache-Superset compatibility with Python 3.12 (bump up numpy dependency to

### DISCUSSION THREAD

**Discussion by Teesmo** [apache/superset#27586] [Q&A / Help]
When I try to install the latest version of Superset (v 3.1.1) I get the following error message:
>AttributeError: module `pkgutil` has no attribute `ImpImporter`. Did you mean: `zipimporter`?
      [end of output]
>  
>note: This error originates from a subprocess, and is likely not a problem with pip.
>error: subprocess-exited-with-error
>
>× Getting requirements to build wheel did not run successfully.
>│ exit code: 1
>╰─> See above for output.
>
>note: This error originates from a subprocess, and is likely not a problem with pip.  

I probed further to discover that it was triggered by an attempt to install `numpy==1.23.5`.
I also tried installing `numpy==1.23.5` itself independently of `apache-superset`, and got the same error. I then did some googling and found that   the attribute in question, `ImpImporter`, has been deprecated since `Python==3.3` and has been removed from `Python==3.12`. Are there plans to move to a more recent version of `numpy` which is compatible to `Python==3.12` (`numpy>=1.26.0`). That is to say, when will `apache-superset` be compatible with `Python==3.12`?

A similar isssue has been raised her [https://github.com/apache/superset/discussions/25963#discussion-5845065](url), but has not been responded to. @nigzak

**Comment by LuvKaranwal21** [comment]
I'm facing the same issue

**Comment by bulmust** [comment]
Some packages like `pandas_gbq` have also dependency conflicts.

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.                                                       
numba 0.57.1 requires numpy<1.25,>=1.21, but you have numpy 2.0.2 which is incompatible.
limits 3.4.0 requires packaging<24,>=21, but you have packaging 25.0 which is incompatible.
apache-superset 3.1.0 requires numpy==1.23.5, but you have numpy 2.0.2 which is incompatible.
```
